core/embed/rust/src/ui/display.rs

```diff
@@ -204,6 +600,131 @@ pub fn set_window(window: Rect) {
     );
 }
 
+pub fn interpolate_colors(color0: Color, color1: Color, step: u16) -> Color {
+    let cr: u16 = ((color0.r() as u16) * step + (color1.r() as u16) * (15 - step)) / 15;
+    let cg: u16 = ((color0.g() as u16) * step + (color1.g() as u16) * (15 - step)) / 15;
+    let cb: u16 = ((color0.b() as u16) * step + (color1.b() as u16) * (15 - step)) / 15;
+    Color::rgb(cr as u8, cg as u8, cb as u8)
+}
+
+pub fn get_color_table(fg_color: Color, bg_color: Color) -> [Color; 16] {
+    let mut table: [Color; 16] = [Color::from_u16(0); 16];
+
+    for (i, item) in table.iter_mut().enumerate() {
+        *item = interpolate_colors(fg_color, bg_color, i as u16);
+    }
+
+    table
+}
+
+pub struct Glyph {
+    width: i32,
+    height: i32,
+    adv: i32,
+    bearing_x: i32,
+    bearing_y: i32,
```

Comment:
they could be, but these are mostly used for computations with other geometry structures like Rect where i32 is used anyway (like in get_pixel function of the TextOverlay), so i chose casting here since it makes the actual computations more readable.